File: src/point4.php
<?php
require_once 'point1.php';
require_once 'point3.php';


$codeTimer = new CodeTimer('Point 4');
$codeTimer->start();

$otherUsersDto = new UsersDto;
try {
    foreach ($data->users as $user) {
        $otherUsersDto = new UserDto($user->id, $user->description);
        throw new \ErrorException('My bad!');
    }

    var_dump(count($usersDto->users));
    
} catch(\Throwable $e) {
    echo $e->getMessage() . "\n";
}

$codeTimer->stop();
echo $codeTimer->result() . "\n";

File: src/point4.py
import point1
import point3

codeTimer = point1.CodeTimer("Point 4")
codeTimer.start()

otherUsersDto = point3.UsersDto()
try:
    for user in point1.data.users:
        userDto = point3.UserDto(user['id'], user['description'])
        raise NameError('My bad!')
except NameError as e:
    print(e)


codeTimer.stop()
print(f'{codeTimer.result()}')

